Clients/GUI/Pages/SignUpWizzard.razor

```diff
@@ -0,0 +1,174 @@
+            {
+                <label style=" color: red;"> @serverErrorMessages </label>
+                <br />
+            }
+
+        </WizardStep>
+        <WizardStep Name="Second Step">
+            <div class="row">
+                <div class="container m-5">
+                    <label>Send your avatar</label>
+                    <br />
+                    <label for="files" class="btn btn-primary">Select Image</label>
+                    <InputFile style="display: none" id="files" accept=".jpg, .jpeg, .png" OnChange="HandleSelection" />
+                    <br />
+                    <label>@fileName</label>
+                    <br />
+                    @if (errorMessageFile.Length != 0)
+                    {
+                        <label style="color: red;"> @errorMessageFile </label>
+                        <br />
+                    }
+                    <button class="btn btn-secondary btn-sm" type="button" @onclick="ClearFile">clear</button>
+                    <br />
+                    @if (isErrorDisplayed)
+                    {
+                        <label style="color: red;"> @serverErrorMessages </label>
+                        <br />
+                    }
+                </div>
+            </div>
+            @if (showLoading)
+            {
+                <div class="spinner-border" role="status">
+                    <span class="sr-only">Loading...</span>
+                </div>
+            }
+        </WizardStep>
+    </Wizard>
+</EditForm>
+
+@code {
+    bool isErrorDisplayed = false;
+    string serverErrorMessages = "";
+    private SignUpViewModel userInput;
+    private EditContext editContext;
+    private string errorMessageFile = "";
+    private string fileName = "";
+    private bool showLoading = false;
+
+    private bool HasError()
+    {
+        return !editContext.Validate() || errorMessageFile.Length != 0;
+    }
+
+    public async Task TrySignUp()
+    {
+        showLoading = true;
+        if (HasError())
+        {
+            showLoading = false;
+            return;
+        }
+        try
+        {
+            var request = new CreateUserRequest
+            {
+                FirstName = userInput.FirstName,
+                LastName = userInput.LastName,
+                Birthday = userInput.Birthday,
+                Email = userInput.Email,
+                Password = userInput.Password,
+                AvatarExtension = userInput.AvatarExtension,
+                Avatar = userInput.Avatar
+            };
+
+            await SignUpper.SignUp(request);
+            NavigationManager.NavigateTo("/signin");
+        }
+        catch (BadRequestException)
+        {
+            isErrorDisplayed = true;
+            serverErrorMessages = "Something went wrong during registration";
+        }
+        catch (NotFoundException)
+        {
+            isErrorDisplayed = true;
+            serverErrorMessages = "Something went wrong during signing you in";
```

Comment:
We can improve this copypasta (turn it into not copypasta)
And also add some more understandable messages instead of _something went wrong_

Reply:
delete useless catch